File: Rabbitool/Config/QQBot.cs
using System.ComponentModel.DataAnnotations;

namespace Rabbitool.Conf;

public class QQBot
{
    [Required]
    public required string AppId { get; set; }

    [Required]
    public required string Token { get; set; }

    [Required]
    public required string SandboxGuildName { get; set; }

    public MarkdownTemplateIds? MarkdownTemplateIds { get; set; }
}

File: conf/qq_bot.go
package conf

type QQBot struct {
	AppId            uint64 `yaml:"appId"`
	Token            string `yaml:"token"`
	IsSandbox        bool   `yaml:"isSandbox"`
	SandboxGuildName string `yaml:"sandboxGuildName"`

	Logger *Logger `yaml:"logger,omitempty"`
}
